services/sqs/src/main/java/software/amazon/awssdk/services/sqs/batchmanager/BatchOverrideConfiguration.java

```diff
@@ -0,0 +1,449 @@
+        result = 31 * result + (maxBatchKeys != null ? maxBatchKeys.hashCode() : 0);
+        result = 31 * result + (maxBufferSize != null ? maxBufferSize.hashCode() : 0);
+        result = 31 * result + (maxBatchOpenDuration != null ? maxBatchOpenDuration.hashCode() : 0);
+        result = 31 * result + (visibilityTimeout != null ? visibilityTimeout.hashCode() : 0);
+        result = 31 * result + (longPollWaitTimeout != null ? longPollWaitTimeout.hashCode() : 0);
+        result = 31 * result + (minReceiveWaitTime != null ? minReceiveWaitTime.hashCode() : 0);
+        result = 31 * result + (messageSystemAttributeNames != null ? messageSystemAttributeNames.hashCode() : 0);
+        result = 31 * result + (receiveMessageAttributeNames != null ? receiveMessageAttributeNames.hashCode() : 0);
+        result = 31 * result + (adaptivePrefetching != null ? adaptivePrefetching.hashCode() : 0);
+        result = 31 * result + (maxInflightReceiveBatches != null ? maxInflightReceiveBatches.hashCode() : 0);
+        result = 31 * result + (maxDoneReceiveBatches != null ? maxDoneReceiveBatches.hashCode() : 0);
+        return result;
+    }
+
+    public static final class Builder implements CopyableBuilder<Builder, BatchOverrideConfiguration> {
+
+        private Integer maxBatchItems;
+        private Integer maxBatchKeys;
+        private Integer maxBufferSize;
+        private Duration maxBatchOpenDuration;
+        private Duration visibilityTimeout;
+        private Duration longPollWaitTimeout;
+        private Duration minReceiveWaitTime;
+        private Integer maxDoneReceiveBatches;
+        private Integer maxInflightReceiveBatches;
+        private List<MessageSystemAttributeName> messageSystemAttributeNames = Collections.emptyList();
+        private List<String> receiveMessageAttributeNames = Collections.emptyList();
+        private Boolean adaptivePrefetching;
+
+        private Builder() {
+        }
+
+        /**
+         * Define the maximum number of messages that are batched together in a single request.
+         *
+         * @param maxBatchItems The new maxBatchItems value.
+         * @return This object for method chaining.
+         */
+        public Builder maxBatchItems(Integer maxBatchItems) {
+            this.maxBatchItems = maxBatchItems;
+            return this;
+        }
+
+        /**
+         * Define the maximum number of batchKeys to keep track of. A batchKey determines which requests are batched together and
+         * is calculated by the client based on the information in a request.
+         * <p>
+         * Ex. SQS determines a batchKey based on a request's queueUrl in combination with its overrideConfiguration, so requests
+         * with the same queueUrl and overrideConfiguration will have the same batchKey and be batched together.
+         *
+         * @param maxBatchKeys the new maxBatchKeys value.
+         * @return This object for method chaining.
+         */
```

Comment:
What happens if the maximum number of batch keys is exceeded? Why is this configurable?